src/nemos/basis.py

```diff
@@ -205,13 +248,23 @@ def __init__(
         *args,
         mode: Literal["eval", "conv"] = "eval",
         window_size: Optional[int] = None,
+        bounds: Optional[Tuple[float, float]] = None,
         **kwargs,
     ) -> None:
         self.n_basis_funcs = n_basis_funcs
         self._n_input_dimensionality = 0
         self._check_n_basis_min()
         self._conv_args = args
         self._conv_kwargs = kwargs
+
+        if bounds is not None and len(bounds) != 2:
+            raise ValueError(
+                f"The provided `bounds` must be of length two. Length {len(bounds)} provided instead!"
+            )
+
+        # convert to float and store
+        self._bounds = bounds if bounds is None else tuple(map(float, bounds))
```

Comment:
presumably this raises an error if the value of bounds aren't numeric. is it a legible error? or should we raise an easy-to-understand one for users?

Reply:
and does all this validation logic belong in a setter?

Reply:
moved to a setter, exception surfaced